lib/build.ts

```diff
@@ -90,20 +91,37 @@ const generateLabels = (task: BuildTask): { labels?: Dictionary<string> } => {
  * @param docker The handle to the docker daemon
  * @return a promise which resolves to a LocalImage which points to the produced image
  */
-export function runBuildTask(
+export async function runBuildTask(
 	task: BuildTask,
 	docker: Dockerode,
 	registrySecrets: RegistrySecrets,
 	secrets?: SecretsPopulationMap,
 	buildArgs?: Dictionary<string>,
 ): Promise<LocalImage> {
```

Comment:
```
Duplicate identifier 'Promise'. Compiler reserves name 'Promise' in top level scope of a module containing async functions.

1 import * as Promise from 'bluebird';
              ~~~~~~~

lib/build.ts:100:4 - error TS1064: The return type of an async function or method must be the global Promise<T> type.

100 ): Promise<LocalImage> {
       ~~~~~~~~~~~~~~~~~~~
```

@pdcastro looks lik it didn't like that I just made the function async

Reply:
Ah, sorry, hadn't realised it was using Bluebird promises instead of native promises - at the top of the file:
```
import * as Promise from 'bluebird';
```
So, either use `.then()` instead of async/await or, probably best, delete the line above (`import * as Promise from 'bluebird';`) so that `Promise` refers to a native promise instead of a Bluebird promise. Up to you!

Reply:
It doesn't look like any Bluebird features are being used - neither inside `runBuildTask` nor by its callers - so it should be safe to delete the `import * as Promise from 'bluebird'` line at the top of the file (which would then allow using `async/await`).